pkg/gadgets/helpers.go

```diff
@@ -196,20 +196,26 @@ var (
 // in the current kernel. False negatives are possible if BTF is not available.
 func DetectBpfKtimeGetBootNs() bool {
 	bpfKtimeGetBootNsOnce.Do(func() {
+		bpfKtimeGetBootNsExists = false
+
 		btfSpec, err := btf.LoadKernelSpec()
 		if err != nil {
-			bpfKtimeGetBootNsExists = false
 			return
 		}
 
 		enum := &btf.Enum{}
 		err = btfSpec.TypeByName("bpf_func_id", &enum)
 		if err != nil {
-			bpfKtimeGetBootNsExists = false
 			return
 		}
 
-		bpfKtimeGetBootNsExists = len(enum.Values) >= BpfKtimeGetBootNsFuncID
+		for _, value := range enum.Values {
+			if value.Value == BpfKtimeGetBootNsFuncID {
```

Comment:
We also need to check the string value, otherwise we'll have an issue similar to the one before. For instance, this check will fail on this specific commit https://github.com/torvalds/linux/blob/0a05861f80fe7d4dcfdabcc98d9854947573e072/include/uapi/linux/bpf.h#L3154, `sk_assign` is 124 and `__BPF_FUNC_MAX_ID` is 125, so this logic will indicate the helper is available when it's not. 

If we want to avoid looping through the array, we can take into consideration that the array is built as:
[0] -> unspec
[1...N-2]: helpers (N-2 being the last helper)
[N-1]: FUNC_MAX_ID

To check if the helper exists we can do `bpfKtimeGetBootNsExists = len(enum.Values) >= BpfKtimeGetBootNsFuncID + 2`
But TBH I think a simple linear search is a lot easier to understand (and to get it right).

Reply:
> I think we can only check the name and ID is not really useful here.

Right, I added that for two (not really important) reasons:
- To speed up a bit the check
- I remembered somebody mentioned that a kernel got a backporting wrong and the ID of a helper was messed up.